gradle-plugin/src/main/kotlin/io/spine/protodata/gradle/Plugin.kt

```diff
@@ -0,0 +1,276 @@
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.Exec
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.kotlin.dsl.withType
+import org.gradle.api.Plugin as GradlePlugin
+
+/**
+ * The ProtoData Gradle plugin.
+ *
+ * Adds the `installProtoData` task which installs the ProtoData executable onto
+ * the current machine.
+ *
+ * Adds the `launchProtoData` task which runs the executable with the arguments assembled from
+ * the configuration of this plugin.
+ *
+ * The users can submit configuration parameters, such as renderer and plugin class names, etc. via
+ * the `protoData { }` extension.
+ *
+ * The users can submit the user classpath to the ProtoData by declaring dependencies using
+ * the `protoData` configuration.
+ *
+ * Example:
+ * ```
+ * protoData {
+ *     renderers("com.acme.MyRenderer")
+ *     plugins("com.acme.MyPlugin")
+ * }
+ *
+ * dependencies {
+ *     protoData(project(":my-plugin"))
+ * }
+ * ```
+ */
+public class Plugin : GradlePlugin<Project> {
+
+    override fun apply(target: Project) {
+        val extension = Extension(target)
+        target.extensions.add("protoData", extension)
+        val protoDataConfiguration = target.configurations.create("protoData")
+        createLaunchTask(target, extension, protoDataConfiguration)
+
+        val resource = Plugin::class.java.classLoader.getResource("version.txt")!!
+        val version = resource.readText()
+        configureProtobufPlugin(target, extension, version)
+
+        val protoDataRawConfiguration = target.configurations.create("protoDataRawArtifact")
+        createInstallTask(target, protoDataRawConfiguration, version)
+    }
+}
+
+/**
+ * The `protoData { }` Gradle extension.
+ */
+@Suppress("UnstableApiUsage") // Gradle Property API.
+public class Extension(private val project: Project) {
+
+    internal  val plugins: ListProperty<String> =
+        project.objects.listProperty(String::class.java)
+
+    /**
+     * Passes given names of Java classes to ProtoData as the `io.spine.protodata.plugin.Plugin`
+     * classes.
+     */
+    public fun plugins(vararg classNames: String) {
+        plugins.addAll(classNames.toList())
+    }
+
+    internal val renderers: ListProperty<String> =
+        project.objects.listProperty(String::class.java)
+
+    /**
+     * Passes given names of Java classes to ProtoData as the `io.spine.protodata.renderer.Renderer`
+     * classes.
+     */
+    public fun renderers(vararg classNames: String) {
+        renderers.addAll(classNames.toList())
+    }
+
+    internal val optionProviders: ListProperty<String> =
+        project.objects.listProperty(String::class.java)
+
+    /**
+     * Passes given names of Java classes to ProtoData as
+     * the `io.spine.protodata.option.OptionsProvider` classes.
+     */
+    public fun optionProviders(vararg classNames: String) {
```

Comment:
Can we have an issue to implement an alternative syntax as discussed?
```
options("this/is/where/my/options.proto", ... )
```

Reply:
Done. See https://github.com/SpineEventEngine/ProtoData/issues/19.